server/real_estate/src/main/java/com/realestate/real_estate/services/UserService.java

```diff
@@ -4,30 +4,45 @@
 import com.realestate.real_estate.repos.BrokerRepository;
 import com.realestate.real_estate.repos.User;
 import com.realestate.real_estate.repos.UserRepository;
-
-import org.bson.types.ObjectId;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataIntegrityViolationException;
 import org.springframework.stereotype.Service;
-
 import java.util.List;
 import java.util.Optional;
 
 @Service
 public class UserService {
-
     @Autowired
     private UserRepository userRepository;
+
+    @Autowired
     private BrokerRepository brokerRepository;
 
     public List<User> getUsers() {
         return userRepository.findAll();
     }
 
-    public User saveUser(User user) {
+    public Optional<Broker> getBroker(String id){
+        return brokerRepository.findById(id);
+    }
+
+    public User signUp(User user) {
+        // Check if user already exists
+        if (userRepository.findByName(user.getName()) != null){
+            throw new DataIntegrityViolationException("The account with username: " + user.getName() + " already exists");
+        }
         return userRepository.save(user);
     }
 
-    public Optional<Broker> getBroker(String id){
-        return brokerRepository.findById(id);
+    public User signIn(String name, String password) {
+        User user = userRepository.findByName(name);
+
+        if (user != null && user.getPassword().equals(password)) {
+            return user;
+        } else {
+            return null; // Authentication failed
```

Comment:
We should return `403 Forbidden Error` here.

Reply:
yeah this give 500, how can we make it return 403

Reply:
ohh not even it 200 but return no body

Reply:
check my comment below